scripts/zones/Wajaom_Woodlands/mobs/Tinnin.lua

```diff
@@ -0,0 +1,161 @@
+        -- First time it regens second head, 25%. Reduced afterwards.
+        if (mob:getLocalVar("secondHead") == 0) then
+            mob:addHP(mob:getMaxHP() * .25);
+            mob:setMod(MOD_REGEN, 60);
+            mob:setLocalVar("secondHead", 1);
+        else
+            mob:addHP(mob:getMaxHP() * .05);
+        end
+        
+    elseif (mob:AnimationSub() == 1 and os.time() > headTimer) then
+        mob:AnimationSub(0);
+        mob:setLocalVar("headTimer", os.time() + math.random(60,190));
+        
+        -- First time it regens third head, 25%. Reduced afterwards.
+        if (mob:getLocalVar("thirdHead") == 0) then
+            mob:addHP(mob:getMaxHP() * .25);
+            mob:setMod(MOD_REGEN, 30);
+            mob:setLocalVar("thirdHead", 1);
+            mob:setUnkillable(false); -- It can be killed now that has all his heads
+        else
+            mob:addHP(mob:getMaxHP() * .05);
+        end
+    end
+end;
+
+-----------------------------------
+-- onMobFight Action
+-----------------------------------
+
+function onMobFight(mob, target)
+    local headTimer = mob:getLocalVar("headTimer");
+    if (mob:AnimationSub() == 2 and os.time() > headTimer) then
+        mob:AnimationSub(1);
+        mob:setMod(MOD_REGEN, 60);
+        mob:setLocalVar("headTimer", os.time() + math.random(60,190));
+        
+        -- First time it regens second head, 25%. Reduced afterwards.
+        if (mob:getLocalVar("secondHead") == 0) then
+            mob:addHP(mob:getMaxHP() * .25);
+            mob:setLocalVar("secondHead", 1);
+        else
+            mob:addHP(mob:getMaxHP() * .05);
+        end        
+        mob:useMobAbility(1576); -- Barofield
+        mob:useMobAbility(1572); -- Pyric Blast
+        
+    elseif (mob:AnimationSub() == 1 and os.time() > headTimer) then
+        mob:AnimationSub(0);
+        mob:setMod(MOD_REGEN, 30);
+        mob:setLocalVar("headTimer", os.time() + math.random(60,190));
+        
+        -- First time it regens third head, 25%. Reduced afterwards.
+        if (mob:getLocalVar("thirdHead") == 0) then
+            mob:addHP(mob:getMaxHP() * .25);
+            mob:setLocalVar("thirdHead", 1);
+            mob:setUnkillable(false); -- It can be killed now that has all his heads
+        else
+            mob:addHP(mob:getMaxHP() * .05);
+        end
+        mob:useMobAbility(1576); -- Barofield
+        mob:useMobAbility(1574); -- Polar Blast
+        
+    end
+end;
+
+-----------------------------------
+-- onCriticalHit
+-----------------------------------
+
+function onCriticalHit(mob)
+    local critNum = mob:getLocalVar("crits");
+    
+    if ((critNum+1) > mob:getLocalVar("CritToTheFace")) then  -- Lose a head
+        if (mob:AnimationSub() == 0) then
+            mob:AnimationSub(1);
+            mob:setMod(MOD_REGEN, 60);
+            mob:setLocalVar("headTimer", os.time() + math.random(60,190));
+        elseif (mob:AnimationSub() == 1) then
+            mob:AnimationSub(2);

```

Comment:
I would need 3 skill lists in the db and it always has access to TP moves. It loses breath attacks when the heads poof.